src/spec-node/containerFeatures.ts

```diff
@@ -129,9 +129,10 @@ export async function getExtendImageBuildInfo(params: DockerResolverParameters,
 	const featuresConfig = await generateFeaturesConfig({ ...params.common, platform }, dstFolder, config.config, getContainerFeaturesFolder, additionalFeatures);
 	if (!featuresConfig) {
 		if (canAddLabelsToContainer && !imageBuildInfo.dockerfile) {
+			const omitDevcontainerPropertyOverride: (keyof DevContainerConfig & keyof ImageMetadataEntry)[] = params.common.skipPersistingRemoteEnvFromConfig ? ['remoteEnv'] : [];
 			return {
 				labels: {
-					[imageMetadataLabel]: JSON.stringify(getDevcontainerMetadata(imageBuildInfo.metadata, config, undefined).raw),
+					[imageMetadataLabel]: JSON.stringify(getDevcontainerMetadata(imageBuildInfo.metadata, config, undefined, [], omitDevcontainerPropertyOverride).raw),
```

Comment:
Two more cases of calls to `getDevcontainerMetadata()` that need the same treatment:
- No features, with Dockerfile: https://github.com/devcontainers/cli/blob/e8493e11c6a920d68fd9b339dbcb7ac77fbf3abf/src/spec-node/containerFeatures.ts#L228
- With features (with or without Dockerfile): https://github.com/devcontainers/cli/blob/e8493e11c6a920d68fd9b339dbcb7ac77fbf3abf/src/spec-node/containerFeatures.ts#L293

Also add tests for these.

I found that following references to `imageMetadataLabel` (from `imageMetadata.ts`) is a good way to verify completeness when reviewing.

Reply:
Great catch. Done